### DISCUSSION TITLE
Arch Testing - `expect('App\Namespace')->unless('App\Namespace\ClassName')`

### DISCUSSION THREAD

**Discussion by nhaynes** [pestphp/pest#1036] [Q&A] [answered]
I'm just getting started with using the Architecture test plugin and I was trying to find an easy way to do the following?

```
arch('app')
    ->expect('Tests\Unit')
    ->unless('Tests\Unit\UnitTestCase') //this is currently not valid
    ->toHaveSuffix('Test');
```

Basically, I want to run expectation on a namespace but provide exceptions to the rule I'm testing for. I think this would be handy for excluding things like `BaseClasses` that you might want organized with the concrete classes but doesn't follow the same naming convention.

Anyone know if this is this currently possible and maybe just missing something here?

**Comment by owenvoke** [comment, marked as answer]
Is [the `ignoring()` method](https://pestphp.com/docs/arch-testing#content-ignoring) what you're looking for?

```php
arch('app')
    ->expect('Tests\Unit')
    ->toHaveSuffix('Test')
    ->ignoring('Tests\Unit\UnitTestCase');

```

**Comment by nhaynes** [comment]
That did the trick! Thank you @owenvoke!
